mbs_results/winsorisation.py

```diff
@@ -0,0 +1,95 @@
+import pandas as pd
+
+from mbs_results.calculate_predicted_unit_value import calculate_predicted_unit_value
+from mbs_results.calculate_ratio_estimation import calculate_ratio_estimation
+from mbs_results.calculate_winsorised_weight import calculate_winsorised_weight
+from mbs_results.flag_for_winsorisation import winsorisation_flag
+
+
+def winsorise(
+    df: pd.DataFrame,
+    group: str,
+    period: str,
+    aux: str,
+    sampled: str,
+    a_weight: str,
+    g_weight: str,
+    target_variable: str,
+    l_values,
+) -> pd.DataFrame:
+
+    """
+    Applies a technique known as one-sided Winsorisation. The objective of the
+    method is to introduce a small bias, while reducing the variance. This is
+    intended to reduce the mean squared error of the total, a measure of
+    overall accuracy.
+
+    The method uses a pre-calculated parameter, 'L-value' that must be
+    supplied to calculate a threshold for each return. The threshold
+    calculated depends upon whether expansion or ratio estimation is used.
```

Comment:
Maybe we should specify ratio estimation is used in this case? What do you think?

Reply:
Agreed